From 1d8c9ed8f56b3fd3e832f3cb8996fa3002edb213 Mon Sep 17 00:00:00 2001
From: Matthias Brugger <mbrugger@suse.com>
Date: Wed, 18 Nov 2015 09:39:39 +0100
Subject: [PATCH] edac: thunderx lmc: Fix the store function
Patch-mainline: Never, this is a fixup for the cavium patch. Should be fixed on mainline.
Reference: fate#319484

The store function didn't return the right value which made the userspace stay
in a loop.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/edac/thunderx_edac-lmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/edac/thunderx_edac-lmc.c b/drivers/edac/thunderx_edac-lmc.c
index 9b5d7ea..eecc9de 100644
--- a/drivers/edac/thunderx_edac-lmc.c
+++ b/drivers/edac/thunderx_edac-lmc.c
@@ -113,7 +113,7 @@ static ssize_t thunderx_lmc_ecc_##_field##_store(struct device *dev,	    \
 	struct thunderx_lmc *lmc = mci->pvt_info;			    \
 	int res = kstrtoull(data, 0, &lmc->_field);			    \
 									    \
-	return res ? count : res;					    \
+	return res ? res : count;					    \
 }									    \
 									    \
 struct device_attribute dev_attr_ecc_##_field =				    \
-- 
2.1.4

